openhcl/openhcl_boot/src/isolation.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Dispatcher for isolation-specific initialization functions
+
+#[cfg(target_arch = "x86_64")]
+use crate::get_tdx_tsc_reftime;
+
+/// Isolation type of the partition
+///
+/// TODO: Fix arch specific abstractions across the bootloader so we can remove
+/// target_arch here and elsewhere.
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+pub enum IsolationType {
+    None,
+    Vbs,
+    #[cfg(target_arch = "x86_64")]
+    Snp,
+    #[cfg(target_arch = "x86_64")]
+    Tdx,
+}
+
+impl IsolationType {
+    pub fn get_ref_time(&self) -> Option<u64> {
```

Comment:
hm i'm not sure i like this function moving here rather than where it was before, since I think we want to remove this whole file in the future (the openhcl_boot only isolationtype).